tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,872 @@
+        },
+
+        #[snafu(display("Failed to create async runtime: {}", source))]
+        Runtime { source: std::io::Error },
+
+        #[snafu(display("Failed to create CFN stack {} in {}: {}", stack_name, region, source))]
+        CreateStack {
+            stack_name: String,
+            region: String,
+            source: rusoto_core::RusotoError<rusoto_cloudformation::CreateStackError>,
+        },
+
+        #[snafu(display(
+            "Failed to fetch stack details for CFN stack {} in {}: {}",
+            stack_name,
+            region,
+            source
+        ))]
+        DescribeStack {
+            stack_name: String,
+            region: String,
+            source: rusoto_core::RusotoError<rusoto_cloudformation::DescribeStacksError>,
+        },
+
+        #[snafu(display("Failed to parse {} to a valid rusoto region: {}", what, source))]
+        ParseRegion {
+            what: String,
+            source: rusoto_core::region::ParseRegionError,
+        },
+
+        #[snafu(display(
+            "Failed to find field {} after attempting to create resource {}",
+            what,
+            resource_name
+        ))]
+        ParseResponse { what: String, resource_name: String },
+
+        #[snafu(display("Failed to get bucket policy for bucket {}: {}", bucket_name, source))]
+        GetPolicy {
+            bucket_name: String,
+            source: rusoto_core::RusotoError<rusoto_s3::GetBucketPolicyError>,
+        },
+
+        #[snafu(display(
+            "Failed to update bucket policy for bucket {}: {}",
+            bucket_name,
+            source
+        ))]
+        PutPolicy {
+            bucket_name: String,
+            source: rusoto_core::RusotoError<rusoto_s3::PutBucketPolicyError>,
+        },
+
+        #[snafu(display("Failed to push object to bucket {}: {}", bucket_name, source))]
+        PutObject {
+            bucket_name: String,
+            source: rusoto_core::RusotoError<rusoto_s3::PutObjectError>,
+        },
+
+        #[snafu(display("Missing environment variable '{}'", var))]
+        Environment {
+            var: String,
+            source: std::env::VarError,
+        },
+
+        #[snafu(display("Failed to read file at {}: {}", path.display(), source))]
+        FileRead { path: PathBuf, source: io::Error },
+
+        #[snafu(display("Failed to open file at {}: {}", path.display(), source))]
+        FileOpen { path: PathBuf, source: io::Error },
+
+        #[snafu(display("Failed to write file in {}: {}", path.display(), source))]
+        FileWrite { path: PathBuf, source: io::Error },
+
+        #[snafu(display(
+            "Must specify both key_alias and regions to generate new keys in Infra.toml, found only one.
+             Please delete all resources created in CloudFormation and re-run create-infra." //QUESTION: this the right place? we can add this as an info! inside the method before error fail 
+        ))]
+        KeyConfig,
```

Comment:
We can change the error display to list out what specifically was missing from the key-config field with a `what` field like the other error types we have.

As for the the "please delete ..." message, I think infrasys needs to be able to clean-up if we fail half-way through to be as atomic as possible. We don't want to leave users with lingering CFN resources that they have to manually delete themselves. I would consider that as another thing to track in an issue somewhere.